engine/transaction/record/testabilities/mock_repository.go

```diff
@@ -1,31 +1,36 @@
-package record_test
+package testabilities
 
 import (
 	"context"
 	"iter"
 	"maps"
 	"slices"
 
-	trx "github.com/bitcoin-sv/go-sdk/transaction"
 	"github.com/bitcoin-sv/spv-wallet/engine/database"
 	"github.com/bitcoin-sv/spv-wallet/models/bsv"
 )
 
-type mockRepository struct {
+type MockRepository struct {
```

Comment:
maybe it doesn't need to be public yet?

Reply:
After my changes (inspired by your comments ☝️) - this requires to be exported (also MockBroadcaster)